### DISCUSSION TITLE
Example code for Mailbox encode and decode functions?

### DISCUSSION THREAD

**Discussion by bradocchs** [pybricks/support#1255] [Q&A] [answered]
Hello,
I'm trying to use the Mailbox class to send bytes between EV3 bricks, but I'm not sure how to properly write the encode and decode functions. Is there any example code for this?
Thanks!

**Comment by dlech** [comment, marked as answer]
If you just need to send single values, you can use the subcalsses like `NumerciMailbox` and `TextMailbox`.

You can look at the [implementation of NumericMailbox](https://github.com/pybricks/pybricks-micropython/blob/04c7304a25b8b16c6cd31ac1caaac501f3fb57be/bricks/ev3dev/modules/pybricks/messaging.py#L105-L116) for inspiration if you want something more complex. It uses [ustruct](https://docs.pybricks.com/en/stable/micropython/ustruct.html). for packing and unpacking.

**Reply by bradocchs** [reply]
Exactly the inspiration I needed! Thanks for the link. Got it working perfectly.
